private/octet-stream.rkt

```diff
@@ -35,38 +36,39 @@
 (define (octet-stream bytes #:padding [padding 0])
   (plain-octet-stream bytes padding))
 
+(define (dissect-octet-stream octets body)
+  (body (octet-stream-bytes octets) (octet-stream-padding octets)))
+
 (define (octet-stream->bitstring octets)
-  (bytes->bitstring (octet-stream-bytes octets)
-                    #:padding (octet-stream-padding octets)))
+  (dissect-octet-stream octets #/fn bytes padding
+  #/bytes->bitstring bytes #:padding padding))
 
 (define (application/octet-stream #:padding [padding 0])
-  (if (zero? padding)
-      (media-type 'application 'octet-stream)
-      (media-type 'application 'octet-stream
-                  #:parameters
-                  (record #:padding (number->immutable-string padding)))))
+  (mat padding 0
+    (media-type 'application 'octet-stream)
+    (media-type 'application 'octet-stream
+                #:parameters
+                (record #:padding #/number->immutable-string padding))))
 
 (define (application/octet-stream? type)
-  (and (equal? (media-type-top-level type) 'application)
-       (equal? (media-type-subtype type) 'octet-stream)
-       (not (media-type-tree type))
-       (not (media-type-suffix type))))
+  (and (equal? 'application #/media-type-top-level type)
+       (equal? 'octet-stream #/media-type-subtype type)
```

Comment:
I think of `(equal? ... 'application)` as an operation that's likely to be abstracted into something like `(application-token? ...)`, and I think of `(equal? (media-type-top-level ...) 'application)` as an operation that's likely to be abstracted into something like `(media-type-is-application? ...)`.

I don't know if those are great choices of name, but at least `equal? 'application` and `equal? 'application #/media-type-top-level`, respectively, are very unambiguous "names." If an abstraction does need to exist, it can focus on having a more evocative name than those, or serve the purpose of being a single place in the code to modify, rather than merely serving as an abbreviation of a common function composition.

Also, I do like to put shorter expressions on the left side. That improves the proximity of an operation to its arguments, especially when some of the shorter arguments fit on the same line as the operation itself.